src/backend/tools/base.py

```diff
@@ -46,19 +46,34 @@ class BaseToolAuthentication:
     Abstract base class for Tool Authentication.
     """
 
-    @classmethod
+    def __init__(self, *args, **kwargs):
+        self.BACKEND_HOST = os.getenv("NEXT_PUBLIC_API_HOSTNAME")
+        self.FRONTEND_HOST = os.getenv("FRONTEND_HOSTNAME")
+        self.AUTH_SECRET_KEY = os.getenv("AUTH_SECRET_KEY")
+
+        self._post_init_check()
+
+    def _post_init_check(self):
+        if any(
```

Comment:
Will this break BE for people who don't set up one of these variables?

Reply:
Great point, I think the GET /tools call inits the tool auth, added a try/catch there